src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -994,32 +1033,7 @@ impl<'gctx> RustcTargetData<'gctx> {
         };
 
         let check_cfg = if gctx.cli_unstable().check_target_cfgs {
-            let mut process = rustc.workspace_process();
```

Comment:
> Use specific check-cfg args if provided for each unit 

This appears to be a refactor combined with a fix?  Could you please split these out so its easier to tell what you are referring to?

Reply:
Also, why are we moving code within a PR rather than making it that way in the first place?

Note: I question moving this as I assume `CheckCfg` would eventually live in `cargo-platform`

Reply:
> Also, why are we moving code within a PR rather than making it that way in the first place?

I wanted to have small commits with clear purpose instead a giant commit implementing all the logic in one go.

Reply:
I believe thats separate from what I'm asking.  The function was added in one commit and then moved in another commit as part of a refactor to unblock feature work (so not quite atomic).  If it was instead in that new location in the original commit, I don't think anything would be negatively affected (if we keep it in that location, which is dependent on other factors).